lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -0,0 +1,488 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedLabels #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- Provides internal functions for the 'UTxOIndex' type, which stores a UTxO
+-- set indexed by asset ID.
+--
+-- See the documentation for 'UTxOIndex' for more details.
+--
+module Cardano.Wallet.Primitive.Types.UTxOIndex.Internal
```

Comment:
Just for the record: in a review call we agreed that leaving it as is would not be harmful.

If there is a need for someone to access the internals of this data type, it would be easy to create a patch that exports the default data constructor.